tools/dist-license/licenses/java/AL 2.0.txt

```diff
@@ -0,0 +1 @@
+https://www.apache.org/licenses/LICENSE-2.0.html
```

Comment:
>Logback is available under EPL 1.0 and LGPL 2.1. The first one is a [category B](https://www.apache.org/legal/resolved#weak-copyleft-licenses) license and Apache projects can include it in **binary** form. The second one is a [category X](https://www.apache.org/legal/resolved#category-x) license and can **not** be included in Apache distributions.

There is an `AND`/`OR` SPDX expression between multiple licenses. An artifact with "category B license OR category X license" is acceptable, but one with "category B license AND category X license" is not.

I have configured this in https://github.com/apache/eventmesh/pull/4827. Since the current version of dependency-review-action only supports `AND` expression, I have to configure some exemptions for now.

>I am not sure if a hyper-link to the license is enough.

This is caused by `amqp-client` misconfiguring the license name, which you can see in the [Maven repository](https://mvnrepository.com/artifact/com.rabbitmq/amqp-client/5.16.0).

For artifacts that do not strictly adhere to the SPDX identifier, the script can only output content according to the information given by the artifact. There should be no need for manual maintenance either.

Reply:
> > I am not sure if a hyper-link to the license is enough.
> 
> This is caused by `amqp-client` misconfiguring the license name, which you can see in the [Maven repository](https://mvnrepository.com/artifact/com.rabbitmq/amqp-client/5.16.0).
> 
> For artifacts that do not strictly adhere to the SPDX identifier, the script can only output content according to the information given by the artifact. There should be no need for manual maintenance either.

I think that the URL to the license is as good an identifier for the license as a SPDX identifier. You can have a lookup table (in `build.gradle` or an external file) that converts URLs to license identifiers.

The lookup table can only be used for misconfigured artifacts. You need a dozen entries at most out of 300 artifacts.

Reply:
Personally, I guess that providing the URL for this license declaration should suffice. Users can access the detailed contents of the license via this URL on devices connected to the internet. Many Apache projects follow a similar approach, such as RocketMQ. However, I will endeavor to download the page pointed to by this URL into the `licenses` folder later on for added reliability.

Reply:
Fixed in https://github.com/apache/eventmesh/pull/4831/commits/356990914ab8917e9c3c98c0f2c55d90282fb495